### PR TITLE
feat: generate and publish mkdocs website

### PR DESCRIPTION
**Pull request by mfrancisc**
This PR should build and deploy the mkdocs website when pushing changes to the `main` branch, we will need to make it nicer and more customizable ;)

kubesawsite

### CODE REVIEW

File: .github/workflows/publish-website.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+name: publish-website
```
Comment (rsoaresd) [inline .github/workflows/publish-website.yml:1]: When following the steps from PR #6, when running `mkdocs serve`, I am getting:

```
(venv) rsoaresd-mac:kubesaw.github.io rsoaresd$ mkdocs serve
WARNING -  Config value 'jobs': Unrecognised configuration name: jobs
WARNING -  Config value 'True': Unrecognised configuration name: True
WARNING -  Config value 'permissions': Unrecognised configuration name: permissions
WARNING -  Config value 'name': Unrecognised configuration name: name
ERROR   -  Config value 'site_name': Required configuration not provided.

Aborted with a configuration error!
```

Do we need this `site_name` or am I missing something?

**Review by alexeykazakov — APPROVED** [review]
Nice!

**Review by MatousJobanek — COMMENTED** [review]
:star_struck: :tada: :clap:

**Review by xcoulon — APPROVED** [review]

**Review by fbm3307 — APPROVED** [review]
👏

**Pull request merged**